mmcv/cnn/bricks/transformer.py

```diff
@@ -0,0 +1,641 @@
+                same shape as `x`. Default None. If None, `x` will be used.
+            query_pos (Tensor): The positional encoding for `query`.
+                Default: None.
+            key_pos (Tensor): The positional encoding for `key`. Default
+                None.
+            reference_points (Tensor):  The normalized reference
+                points with shape (bs, num_query, num_levels, 2),
+                all elements is range in [0, 1], top-left (0,0),
+                bottom-right (1, 1), including padding area.
+                or (N, Length_{query}, num_levels, 4), add
+                additional two dimensions is (w, h) to
+                form reference boxes.
+            key_padding_mask (Tensor): ByteTensor for `query`, with
+                shape [bs, num_key].
+            spatial_shapes (Tensor): Spatial shape of features in
+                different level. With shape  (num_levels, 2),
+                last dimension represent (h, w).
+            level_start_index (Tensor): The start index of each level.
+                A tensor has shape (num_levels) and can be represented
+                as [0, h_0*w_0, h_0*w_0+h_1*w_1, ...].
+
+        Returns:
+             Tensor: forwarded results with shape [num_query, bs, embed_dims].
+        """
+
+        if key is None:
+            key = query
+        if value is None:
+            value = key
+        else:
+            assert value is key or key is None, 'In the implementation of' \
+                'MultiScaleDeformableAttention, value is key.'
+        assert residual is None, 'MultiScaleDeformableAttention dose not' \
+                                 'support add residual'
+        if query_pos is not None:
+            query = query + query_pos
+
+        # change to (bs, num_query ,embed_dims)
+        query = query.permute(1, 0, 2)
+        value = value.permute(1, 0, 2)
+
+        bs, num_query, _ = query.shape
+        bs, num_key, _ = value.shape
+        assert (spatial_shapes[:, 0] * spatial_shapes[:, 1]).sum() == num_key
+
+        value = self.value_proj(value)
+        if key_padding_mask is not None:
+            value = value.masked_fill(key_padding_mask[..., None], 0.0)
+        value = value.view(bs, num_key, self.num_heads, -1)
+        sampling_offsets = self.sampling_offsets(query).view(
+            bs, num_query, self.num_heads, self.num_levels, self.num_points, 2)
+        attention_weights = self.attention_weights(query).view(
+            bs, num_query, self.num_heads, self.num_levels, self.num_points)
+        attention_weights = attention_weights.softmax(-1)
+        # N, num_query, num_heads, num_levels, num_points, 2
+        if reference_points.shape[-1] == 2:
+            offset_normalizer = torch.stack(
+                [spatial_shapes[..., 1], spatial_shapes[..., 0]], -1)
+            sampling_locations = reference_points[:, :, None, :, None, :] + \
+                sampling_offsets / \
```

Comment:
This format breaks pep8 recommendation.
It is recommanded to use `/ offsetxxxxx` rather than `sampling_offsets / \`
see https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator